File: components/checkbox/flutter/ciata_checkbox.dart
import 'package:flutter/material.dart';

/// Implementação experimental do CMP-0003 Checkbox para Flutter.
class CiataCheckbox extends StatelessWidget {
  CiataCheckbox({
    super.key,
    required String label,
    required this.value,
    required this.onChanged,
    this.requiredField = false,
    this.enabled = true,
    String? helpText,
    String? errorText,
    this.tristate = false,
  })  : label = label.trim(),
        helpText = helpText?.trim(),
        errorText = errorText?.trim() {
    if (this.label.isEmpty) {
      throw ArgumentError.value(label, 'label', 'não pode ser vazio');
    }
    if (!tristate && value == null) {
      throw ArgumentError.value(value, 'value', 'não pode ser null quando tristate=false');
    }
  }

  final String label;
  final bool? value;
  final ValueChanged<bool?>? onChanged;
  final bool requiredField;
  final bool enabled;
  final String? helpText;
  final String? errorText;
  final bool tristate;

  @override
  Widget build(BuildContext context) {
    final effectiveOnChanged = enabled ? onChanged : null;
    final visibleLabel = requiredField ? '$label (obrigatório)' : label;

    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        CheckboxListTile(
          value: value,
          tristate: tristate,
          onChanged: effectiveOnChanged,
          title: Text(visibleLabel),
          controlAffinity: ListTileControlAffinity.leading,
          contentPadding: EdgeInsets.zero,
        ),
        if (helpText case final text? when text.isNotEmpty)
          Text(text, style: Theme.of(context).textTheme.bodySmall),
        if (errorText case final text? when text.isNotEmpty)
          Semantics(
            liveRegion: false,
            label: 'Erro: $text',
            child: Text(
              text,
              style: Theme.of(context).textTheme.bodySmall?.copyWith(
                    color: Theme.of(context).colorScheme.error,
                  ),
            ),
          ),
      ],
    );
  }
}

File: components/checkbox/ios/CiataCheckbox.swift
import SwiftUI

/// Implementação experimental do CMP-0003 Checkbox para SwiftUI.
public struct CiataCheckbox: View {
    private let label: String
    @Binding private var isChecked: Bool
    private let required: Bool
    private let disabled: Bool
    private let helpText: String?
    private let errorText: String?

    public init(
        _ label: String,
        isChecked: Binding<Bool>,
        required: Bool = false,
        disabled: Bool = false,
        helpText: String? = nil,
        errorText: String? = nil
    ) {
        precondition(!label.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty, "label não pode ser vazio")
        self.label = label
        self._isChecked = isChecked
        self.required = required
        self.disabled = disabled
        self.helpText = helpText
        self.errorText = errorText
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            Toggle(isOn: $isChecked) {
                Text(required ? "\(label) (obrigatório)" : label)
            }
            .toggleStyle(.switch)
            .disabled(disabled)
            .accessibilityHint(accessibilityHint)

            if let helpText, !helpText.isEmpty {
                Text(helpText)
                    .font(.footnote)
                    .foregroundStyle(.secondary)
            }

            if let errorText, !errorText.isEmpty {
                Text(errorText)
                    .font(.footnote)
                    .accessibilityLabel("Erro: \(errorText)")
            }
        }
    }

    private var accessibilityHint: String {
        [helpText, errorText.map { "Erro: \($0)" }]
            .compactMap { $0 }
            .filter { !$0.isEmpty }
            .joined(separator: ". ")
    }
}
